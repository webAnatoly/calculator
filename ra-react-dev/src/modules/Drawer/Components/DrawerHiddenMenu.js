import React, { Component } from 'react'
import Divider from 'material-ui/Divider'
import List from 'material-ui/List'
import { withStyles } from 'material-ui/styles'
import { styles } from '../theme'
import DrawerMenuItem from './DrawerMenuItem'

export class HiddenBlock extends Component {
  render () {
    const { opened, toggleDrawer, items } = this.props

    return (
      <div>
        <List onClick={() => (opened ? toggleDrawer() : null)}>
          {items.map(item => (
            <DrawerMenuItem key={`hidden_${item.label}`} {...item} />
          ))}
        </List>
        <Divider />
      </div>
    )
  }
}

export default withStyles(styles, { withTheme: true })(HiddenBlock)
