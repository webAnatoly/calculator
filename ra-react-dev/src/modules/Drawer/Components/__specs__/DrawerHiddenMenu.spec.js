import React from 'react'
import { shallow } from 'enzyme'
import { HiddenBlock } from '../DrawerHiddenMenu'

describe('DrawerHiddenMenu', () => {
  let toggleDrawer, items, opened, wrapper
  beforeEach(() => {
    toggleDrawer = jest.fn()
    items = [{ label: 'testItem1' }]
    opened = true
    wrapper = shallow(
      <HiddenBlock opened={opened} toggleDrawer={toggleDrawer} items={items} />
    )
  })

  it('should renders as expected', () => {
    expect(wrapper).toMatchSnapshot()
  })

  it('should handle onClick', () => {
    wrapper
      .childAt(0)
      .props()
      .onClick()
    expect(toggleDrawer.mock.calls.length).toEqual(1)
    wrapper.setProps({ opened: false })
    wrapper
      .childAt(0)
      .props()
      .onClick()
    expect(toggleDrawer.mock.calls.length).toEqual(1)
  })
})
