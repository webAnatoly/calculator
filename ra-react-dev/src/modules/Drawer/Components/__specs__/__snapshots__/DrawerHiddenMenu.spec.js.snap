// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`DrawerHiddenMenu should renders as expected 1`] = `
<div>
  <withStyles(List)
    onClick={[Function]}
  >
    <DrawerMenuItem
      key="hidden_testItem1"
      label="testItem1"
    />
  </withStyles(List)>
  <withStyles(Divider) />
</div>
`;
